exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -18,111 +18,131 @@ import (
+	}
 
-	seqTokenMu sync.Mutex
-	seqToken   string
-}
+	expConfig := config.(*Config)
+	expConfig.logger = params.Logger
 
-func (e *exporter) Start(ctx context.Context, host component.Host) error {
-	var startErr error
-	e.startOnce.Do(func() {
-		awsConfig := &aws.Config{}
-		if e.config.Region != "" {
-			awsConfig.Region = aws.String(e.config.Region)
-		}
-		if e.config.Endpoint != "" {
-			awsConfig.Endpoint = aws.String(e.config.Endpoint)
-		}
-		awsConfig.MaxRetries = aws.Int(1) // retry will be handled by the collector queue
-		sess, err := session.NewSession(awsConfig)
-		if err != nil {
-			startErr = err
-			return
-		}
-		e.client = cloudwatchlogs.New(sess)
+	// create AWS session
+	awsConfig, session, err := awsutil.GetAWSConfigSession(params.Logger, &awsutil.Conn{}, &expConfig.AWSSessionSettings)
+	if err != nil {
+		return nil, err
+	}
 
-		e.logger.Debug("Retrieving CloudWatch sequence token")
-		out, err := e.client.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
-			LogGroupName:        aws.String(e.config.LogGroupName),
-			LogStreamNamePrefix: aws.String(e.config.LogStreamName),
-		})
-		if err != nil {
-			startErr = err
-			return
-		}
-		if len(out.LogStreams) == 0 {
-			startErr = errors.New("cannot find log group and stream")
-			return
-		}
-		stream := out.LogStreams[0]
-		if stream.UploadSequenceToken == nil {
-			e.logger.Debug("CloudWatch sequence token is nil, will assume empty")
-			return
-		}
-		e.seqToken = *stream.UploadSequenceToken
-	})
-	return startErr
-}
+	// create CWLogs client with aws session config
+	svcStructuredLog := cwlogs.NewClient(params.Logger, awsConfig, params.BuildInfo, expConfig.LogGroupName, session)
+	collectorIdentifier, err := uuid.NewRandom()
 
-func (e *exporter) Shutdown(ctx context.Context) error {
-	// TODO(jbd): Signal shutdown to flush the logs.
-	return nil
+	if err != nil {
+		return nil, err
+	}
+
+	expConfig.Validate()
+
+	logsExporter := &exporter{
+		svcStructuredLog: svcStructuredLog,
+		config:           config,
+		logger:           params.Logger,
+		retryCount:       *awsConfig.MaxRetries,
+		collectorID:      collectorIdentifier.String(),
+	}
+	return logsExporter, nil
 }
 
-func (e *exporter) PushLogs(ctx context.Context, ld pdata.Logs) (err error) {
-	// TODO(jbd): Relax this once CW Logs support ingest
-	// without sequence tokens.
-	e.seqTokenMu.Lock()
-	defer e.seqTokenMu.Unlock()
+func newCwLogsExporter(config config.Exporter, params component.ExporterCreateSettings) (component.LogsExporter, error) {
+	logsExporter, err := newCwLogsPusher(config, params)
+	if err != nil {
+		return nil, err
+	}
+	expConfig := config.(*Config)
+	return exporterhelper.NewLogsExporter(
+		config,
+		params,
+		logsExporter.ConsumeLogs,
+		exporterhelper.WithQueue(expConfig.enforcedQueueSettings()),
+		exporterhelper.WithRetry(expConfig.RetrySettings),
+	)
+
+}
 
+func (e *exporter) ConsumeLogs(ctx context.Context, ld pdata.Logs) error {
+	exp := e.config.(*Config)
```

Comment:
Eliminating `getLogPusher()` will also eliminate the need for this coercion, but I would suggest storing the `*Config` directly on the exporter rather than the `config.Exporter` interface.  Then you should only need to make this coercion once, instead of the three or more times it is currently done.